core/src/main/java/org/elasticsearch/discovery/zen/ZenDiscovery.java

```diff
@@ -634,9 +624,7 @@ public ClusterState execute(ClusterState currentState) {
                         .masterNodeId(null).build();
 
                 // flush any pending cluster states from old master, so it will not be set as master again
-                ArrayList<ProcessClusterState> pendingNewClusterStates = new ArrayList<>();
-                processNewClusterStates.drainTo(pendingNewClusterStates);
-                logger.trace("removed [{}] pending cluster states", pendingNewClusterStates.size());
+                publishClusterState.pendingStatesQueue().failAllStatesAndClear(new ElasticsearchException("master left [" + reason + "]"));
```

Comment:
`ElasticsearchException` supports the `{}` formatting like logging does, using that we can remove the string concatenation (and subsequent extra object creation) here.